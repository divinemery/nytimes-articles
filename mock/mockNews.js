export default [
    {
        "url": "https://www.nytimes.com/2019/07/14/us/politics/trump-twitter-squad-congress.html",
        "adx_keywords": "Trump, Donald J;United States Politics and Government;Omar, Ilhan;Ocasio-Cortez, Alexandria;Tlaib, Rashida;Pressley, Ayanna;Pelosi, Nancy;Discrimination;Immigration and Emigration;House of Representatives;Somali-Americans",
        "column": null,
        "section": "U.S.",
        "byline": "By KATIE ROGERS and NICHOLAS FANDOS",
        "type": "Article",
        "title": "Trump Tells Congresswomen to ‘Go Back’ to the Countries They Came From",
        "abstract": "The attack, which was widely established as a racist trope, appeared to be meant for four progressive freshmen who have clashed with Speaker Nancy Pelosi. Only one was born outside the United States.",
        "published_date": "2019-07-14",
        "source": "The New York Times",
        "id": 100000006610933,
        "asset_id": 100000006610933,
        "views": 2,
        "des_facet": [
            "UNITED STATES POLITICS AND GOVERNMENT",
            "IMMIGRATION AND EMIGRATION",
            "SOMALI-AMERICANS"
        ],
        "org_facet": [
            "DISCRIMINATION",
            "HOUSE OF REPRESENTATIVES"
        ],
        "per_facet": [
            "TRUMP, DONALD J",
            "OMAR, ILHAN",
            "OCASIO-CORTEZ, ALEXANDRIA",
            "TLAIB, RASHIDA",
            "PRESSLEY, AYANNA",
            "PELOSI, NANCY"
        ],
        "geo_facet": "",
        "media": [
            {
                "type": "image",
                "subtype": "photo",
                "caption": "House Democrats rallied around Representatives Alexandria Ocasio-Cortez of New York, left, Rashida Tlaib of Michigan, center, and Ayanna S. Pressley of Massachusetts.",
                "copyright": "Anna Moneymaker/The New York Times",
                "approved_for_syndication": 1,
                "media-metadata": [
                    {
                        "url": "https://static01.nyt.com/images/2019/07/14/us/politics/14dc-trump-hp-promo/14dc-trump-hp-promo-thumbStandard.jpg",
                        "format": "Standard Thumbnail",
                        "height": 75,
                        "width": 75
                    },
                    {
                        "url": "https://static01.nyt.com/images/2019/07/14/us/politics/14dc-trump-hp-promo/14dc-trump-hp-promo-mediumThreeByTwo210.jpg",
                        "format": "mediumThreeByTwo210",
                        "height": 140,
                        "width": 210
                    },
                    {
                        "url": "https://static01.nyt.com/images/2019/07/14/us/politics/14dc-trump-hp-promo/14dc-trump-hp-promo-mediumThreeByTwo440.jpg",
                        "format": "mediumThreeByTwo440",
                        "height": 293,
                        "width": 440
                    }
                ],
            }
        ],
        "uri": "nyt://article/0536b0da-56f4-5ef1-bed9-93bd18f20ba5"
    }
]