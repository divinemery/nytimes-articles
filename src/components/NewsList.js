import { Skeleton, Switch, List, Avatar, Icon } from 'antd';
import styles from './newslist.css';
import router from 'umi/router';


const NewsList = ({ news }) => {
  const IconText = ({ type, text }) => (
    <span>
      <Icon type={type} style={{ marginRight: 8 }} />
      {text}
    </span>
  );

  let SkeletonList = [];
  for (let i = 0; i < 5; i++) {
    SkeletonList.push(<Skeleton key={'index-'+i} className={styles.skeletonElt} active avatar /> );
  }

  if (news.loading) {
    return SkeletonList;
  } else {
    return (
      <List
        itemLayout="vertical"
        size="large"
        dataSource={news.data}
        renderItem={item => (
          <List.Item
          // onClick={()=> {  router.push('/newsArticle')}}
          key={item.id}
          extra={
            <div className={styles.imageBlock}>
            <img
            className={styles.imageItem}
            alt="logo"
            src={item['media'][0]['media-metadata'][1].url}
            />
            <p className={styles.imageCaption}><em>{item['media'][0]['caption']}</em></p>  
            </div>          
          }
          >
            <div className={styles.articleSection}>#{item.section}</div>
            <List.Item.Meta
              className={styles.articleTitle}              
              title={<a href={item.url} target="_blank">{item.title}</a>}
              description={item.abstract}
            />
            <div className={styles.byLineStyle}>
              {item.views} views  
              <span className={styles.verticalSpace}>|</span> 
              {item.byline} 
              <span className={styles.verticalSpace}>|</span>
              On {item.published_date}
            </div>
            <div className={styles.viewStyle}>
            </div>
          </List.Item>
        )}
      />
    );
  }
};

export default NewsList;
