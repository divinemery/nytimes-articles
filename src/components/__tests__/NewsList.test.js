import NewsList  from '../NewsList.js';
import renderer from 'react-test-renderer';
import mockNews from '../../../mock/mockNews';

describe('Component: NewsList', () => {
  it('Render correctly', () => {    
    let mockData1 = {
      data: [],
      total: 0,
      loading: false,
    }, mockData2 = {
      data: [],
      total: 0,
      loading: true,
    };

    let mockData3 = {
      data: mockNews,
      total: 0,
      loading: true
    };

    const loadingCmp = renderer.create(<NewsList news={mockData1} />);
    const emptyDataCmp = renderer.create(<NewsList news={mockData2} />);
    const dataCmp = renderer.create(<NewsList news={mockData3} />);

    let tree = loadingCmp.toJSON();
    expect(tree).toMatchSnapshot();

    tree = emptyDataCmp.toJSON();
    expect(tree).toMatchSnapshot();

    tree = dataCmp.toJSON();
    expect(tree).toMatchSnapshot();

  });
});
