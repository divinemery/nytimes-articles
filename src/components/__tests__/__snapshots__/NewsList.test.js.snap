// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: NewsList Render correctly 1`] = `
<div
  className="ant-list ant-list-vertical ant-list-lg ant-list-split"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-list-empty-text"
      >
        <div
          className="ant-empty ant-empty-normal"
        >
          <div
            className="ant-empty-image"
          >
            <img
              alt="No Data"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHZpZXdCb3g9IjAgMCA2NCA0MSIgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
            />
          </div>
          <p
            className="ant-empty-description"
          >
            No Data
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component: NewsList Render correctly 2`] = `
Array [
  <div
    className="ant-skeleton skeletonElt ant-skeleton-with-avatar ant-skeleton-active"
  >
    <div
      className="ant-skeleton-header"
    >
      <span
        className="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
        style={Object {}}
      />
    </div>
    <div
      className="ant-skeleton-content"
    >
      <h3
        className="ant-skeleton-title"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <ul
        className="ant-skeleton-paragraph"
      >
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
      </ul>
    </div>
  </div>,
  <div
    className="ant-skeleton skeletonElt ant-skeleton-with-avatar ant-skeleton-active"
  >
    <div
      className="ant-skeleton-header"
    >
      <span
        className="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
        style={Object {}}
      />
    </div>
    <div
      className="ant-skeleton-content"
    >
      <h3
        className="ant-skeleton-title"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <ul
        className="ant-skeleton-paragraph"
      >
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
      </ul>
    </div>
  </div>,
  <div
    className="ant-skeleton skeletonElt ant-skeleton-with-avatar ant-skeleton-active"
  >
    <div
      className="ant-skeleton-header"
    >
      <span
        className="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
        style={Object {}}
      />
    </div>
    <div
      className="ant-skeleton-content"
    >
      <h3
        className="ant-skeleton-title"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <ul
        className="ant-skeleton-paragraph"
      >
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
      </ul>
    </div>
  </div>,
  <div
    className="ant-skeleton skeletonElt ant-skeleton-with-avatar ant-skeleton-active"
  >
    <div
      className="ant-skeleton-header"
    >
      <span
        className="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
        style={Object {}}
      />
    </div>
    <div
      className="ant-skeleton-content"
    >
      <h3
        className="ant-skeleton-title"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <ul
        className="ant-skeleton-paragraph"
      >
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
      </ul>
    </div>
  </div>,
  <div
    className="ant-skeleton skeletonElt ant-skeleton-with-avatar ant-skeleton-active"
  >
    <div
      className="ant-skeleton-header"
    >
      <span
        className="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
        style={Object {}}
      />
    </div>
    <div
      className="ant-skeleton-content"
    >
      <h3
        className="ant-skeleton-title"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <ul
        className="ant-skeleton-paragraph"
      >
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
      </ul>
    </div>
  </div>,
]
`;

exports[`Component: NewsList Render correctly 3`] = `
Array [
  <div
    className="ant-skeleton skeletonElt ant-skeleton-with-avatar ant-skeleton-active"
  >
    <div
      className="ant-skeleton-header"
    >
      <span
        className="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
        style={Object {}}
      />
    </div>
    <div
      className="ant-skeleton-content"
    >
      <h3
        className="ant-skeleton-title"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <ul
        className="ant-skeleton-paragraph"
      >
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
      </ul>
    </div>
  </div>,
  <div
    className="ant-skeleton skeletonElt ant-skeleton-with-avatar ant-skeleton-active"
  >
    <div
      className="ant-skeleton-header"
    >
      <span
        className="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
        style={Object {}}
      />
    </div>
    <div
      className="ant-skeleton-content"
    >
      <h3
        className="ant-skeleton-title"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <ul
        className="ant-skeleton-paragraph"
      >
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
      </ul>
    </div>
  </div>,
  <div
    className="ant-skeleton skeletonElt ant-skeleton-with-avatar ant-skeleton-active"
  >
    <div
      className="ant-skeleton-header"
    >
      <span
        className="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
        style={Object {}}
      />
    </div>
    <div
      className="ant-skeleton-content"
    >
      <h3
        className="ant-skeleton-title"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <ul
        className="ant-skeleton-paragraph"
      >
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
      </ul>
    </div>
  </div>,
  <div
    className="ant-skeleton skeletonElt ant-skeleton-with-avatar ant-skeleton-active"
  >
    <div
      className="ant-skeleton-header"
    >
      <span
        className="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
        style={Object {}}
      />
    </div>
    <div
      className="ant-skeleton-content"
    >
      <h3
        className="ant-skeleton-title"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <ul
        className="ant-skeleton-paragraph"
      >
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
      </ul>
    </div>
  </div>,
  <div
    className="ant-skeleton skeletonElt ant-skeleton-with-avatar ant-skeleton-active"
  >
    <div
      className="ant-skeleton-header"
    >
      <span
        className="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
        style={Object {}}
      />
    </div>
    <div
      className="ant-skeleton-content"
    >
      <h3
        className="ant-skeleton-title"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <ul
        className="ant-skeleton-paragraph"
      >
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
      </ul>
    </div>
  </div>,
]
`;
